### PR TITLE
Order

### PR DESCRIPTION
**Pull request by dtingg**
Added some tests for order.
Added github photos.
Added more links for photos and merchant names.
Added methods for orderitem.total and order.total

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
       redirect_back(fallback_location: root_path)
       return
     end
     
-    if @order.update(order_params)
+    if @cart.update(order_params)
       flash[:success] = "Thank you for your order!"  
-      redirect_to order_path(@order)
+      redirect_to order_path(@cart)
       session[:order_id] = nil
```
Comment (NatalieTapias) [inline app/controllers/orders_controller.rb:34]: nice

File: app/views/orders/cart.html.erb [hunk 2]
```diff
@@ -1,43 +1,46 @@
...
+    <td><%= link_to "#{item.product.merchant.username}", merchant_path(item.product.merchant) %></td>
+    <td><%= form_with model: item do |f| %>
+      <%= f.hidden_field :id, value: @cart.id %>
+      <%= hidden_field_tag :old_quantity, item.quantity %>
+      <% available = item.product.stock_qty + item.quantity %>
+      <%= f.select :quantity, (1..available)%>
+      <%= f.submit "Update quantity" %>
+    <% end %></td>
+    <td>$<%= sprintf('%.2f', item.product.price) %></td>
+    <td>$<%= sprintf('%.2f', (item.quantity * item.product.price)) %></td>
```
Comment (NatalieTapias) [inline app/views/orders/cart.html.erb:28]: With more time we can pull some of this logic out of the view and into view helper file

File: app/views/products/index.html.erb [hunk 3]
```diff
@@ -19,7 +19,8 @@
   </section>
 
   <section class="products-list">
-    <% Product.order_active_products.each do |product| %>
+    <%# <% Product.order_active_products.each do |product| %>
+    <% @products.where(active: true).where("stock_qty > ?", 0).each do |product| %>
```
Comment (NatalieTapias) [inline app/views/products/index.html.erb:23]: We want to pull any model logic in the view out of the view

File: app/views/orders/cart.html.erb (hunk 2 above)
Comment (KKennedyCodes) [reply app/views/orders/cart.html.erb:28]: I second that.

**Review by KKennedyCodes — APPROVED** [review]

**Pull request merged**